src/Factory.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Part of the Joomla Framework Registry Package
+ *
+ * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Registry;
+
+/**
+ * Factory class to fetch Registry objects
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Factory
+{
+	/**
+	 * Format instances container - for backward compatibility with AbstractRegistryFormat::getInstance().
+	 *
+	 * @var    array
+	 * @since  __DEPLOY_VERSION__
+	 * @deprecated  2.0  Object caching will no longer be supported
+	 */
+	protected static $formatInstances = array();
+
+	/**
+	 * Returns a AbstractRegistryFormat object, only creating it if it doesn't already exist.
+	 *
+	 * @param   string  $type     The format to load
+	 * @param   array   $options  Additional options to configure the object
+	 *
+	 * @return  AbstractRegistryFormat  Registry format handler
```

Comment:
I'd say that's crap code being perfectly blunt.  If you can't rely on a factory method returning a class implementing a documented interface, then why bother having the documentation or factory in general?  Is it really the job of our factories to say "you can't instantiate this class because it doesn't implement the interface"?  No platform I've got any familiarity with does this but Joomla.